openff/interchange/interop/gromacs/export/_export.py

```diff
@@ -60,31 +60,71 @@ def _write_defaults(self, top):
             f"{self.system.coul_14:8.6f}\n\n",
         )
 
-    def _write_atomtypes(self, top):
+    def _write_atomtypes(self, top) -> dict:
         top.write("[ atomtypes ]\n")
         top.write(
             ";type, bondingtype, atomic_number, mass, charge, ptype, sigma, epsilon\n",
         )
 
+        reduced_atom_types = []
+        to_reduced_atom_types = {}
+
+        def _is_atom_type_in_list(
+            atom_type, atom_type_list, diff: float = 1e-5
+        ) -> bool | str:
+            """
+            Checks if the atom type is already in list.
+            """
+            for _at_name, _atom_type in atom_type_list:
+                if (
+                    atom_type.atomic_number == _atom_type.atomic_number
+                    and numpy.abs(atom_type.mass.m - _atom_type.mass.m) < diff
+                    and numpy.abs(atom_type.sigma.m - _atom_type.sigma.m) < diff
+                    and numpy.abs(atom_type.epsilon.m - _atom_type.epsilon.m) < diff
```

Comment:
Since these aren't vectorized, and since in the future they might be stored with different units, I think something like this might be safer and/or faster

```suggestion
                    and abs(atom_type.mass - _atom_type.mass).m < mass_tolerance
                    and abs(atom_type.sigma - _atom_type.sigma).m < sigma_tolerance
                    and abs(atom_type.epsilon - _atom_type.epsilon).m < epsilon_tolerance
```
with something like

```python
mass_tolerance = Quantity("1e-5 amu")
sigma_tolerance = Quantity("1e-5 nanometer")
epsilon_tolerance = Quantity("1e-5 kilojoule_per_mol")
```
defined above (to avoid re-initializing each during every loop)